### PR TITLE
tests(gatsby): Create docker container and test site for future memory usage tests and benchmarks

### PR DESCRIPTION
**Pull request by j0sh77**
## Description
The goal of this new benchmark test site is to set up an environment we can use to work through memory optimizations and scaling issues. Within this test site, we currently create nodes of large size to store in memory, which should help us trigger OOM issues and work through solutions in the garbage collector and LMDB cache. 

To simulate a low-resource environment, we're using docker with a default of 2GB of memory. At this time, it's only configurable by changing the `docker:run` command, but we may make it configurable in the future.

This site is also set up to work with the Chrome debugger from within Docker! 🥳 

### Documentation

See the added [README](https://github.com/gatsbyjs/gatsby/blob/memory-test-site/benchmarks/memory/README.md)! This has all of the available yarn commands and a summary of this site.

### CODE REVIEW

File: benchmarks/memory/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Runs `gatsby develop` with `inspect-brk` set to start the [debugging process](https://www.gatsbyjs.com/docs/debugging-the-build-process/) on port 9229.
+
+## Setup
+
+Currently we can reproduce builds crashing with out default settings
+
+- Docker container running with 2GB limit
+- 300 nodes x ~2MB each = ~600MB of "just" nodes data in each process (number of nodes can be controlled with NUM_NODES env var)
+- 3 workers + main process (GATSBY_CPU_COUNT set to 4 in docker image, but you can specify different value with env var)
```
[the remark this answers is not in the document]
Comment (pieh) [reply benchmarks/memory/README.md:71]: I'll reword this, but essentially I meant `GATSBY_CPU_COUNT=6 gatsby build` (for now we are not automating this to run in CI and only using it as test site / environment to test changes in limited mem environment)

[the remark this answers is not in the document]
Comment (pieh) [reply benchmarks/memory/README.md]: is [`aae48ed` (#34595)](https://github.com/gatsbyjs/gatsby/pull/34595/commits/aae48edcae7d497d3ac5f1e8e41dd4e17cf4f627) better?

[the remark this answers is not in the document]
Comment (LekoArts) [reply benchmarks/memory/README.md]: Looks good 👍

**Review by LekoArts — APPROVED** [review]
LGTM

**Pull request merged**